docstore/gcpfirestore/fs.go

```diff
@@ -554,31 +554,44 @@ func (c *collection) doCommitCall(ctx context.Context, call *commitCall, errs []
 		}
 		return
 	}
-	_ = wrs
 	// Set the revision fields of the documents.
 	// The actions and writes may not correspond, because Update actions may require
 	// two writes. We can tell which writes correspond to actions by the type of write.
 	j := 0
-	for i, wr := range wrs {
-		if _, ok := call.writes[i].Operation.(*pb.Write_Transform); !ok {
-			a := call.actions[j]
-			if a.Doc.HasField(c.opts.RevisionField) {
-				if err := a.Doc.SetField(c.opts.RevisionField, wr.UpdateTime); err != nil {
-					errs[a.Index] = err
-				}
+	for i, a := range call.actions {
```

Comment:
Yes, the old way causes updates with increment only mods to fail. I updated the PR description about the details.

This change combines the two `if` together, so it would be about the same amount of code either way, it would be that `hasFollowingTransform` becomes something like a `hasLeadingUpdate` check.